src/serde/writer.rs

```diff
@@ -29,17 +29,17 @@ pub enum SerializeError {
 ///
 /// This struct should not be used often.
 /// For most cases, prefer the `encode_into` function.
-pub struct Serializer<'a, W: 'a> {
-    writer: &'a mut W,
+pub struct Serializer<W> {
+    writer: W,
 }
 
 fn wrap_io(err: IoError) -> SerializeError {
     SerializeError::IoError(err)
 }
 
 impl serde::ser::Error for SerializeError {
-    fn custom<T: Into<String>>(msg: T) -> Self {
-        SerializeError::Custom(msg.into())
+    fn custom<T: ::std::fmt::Display>(msg: T) -> Self {
+        SerializeError::Custom(format!("{}", msg))
```

Comment:
I think this would be clearer as `msg.to_string()` instead of `format!("{}", msg)`.